honeybadgermpc/batch_reconstruction.py

```diff
@@ -118,51 +150,59 @@ def list_transpose(lst):
       the reconstructed array of B shares
 
     Communication takes place over two rounds,
-      objects sent/received of the form ('R1', shares) or ('R2', shares)
+      objects sent/received of the form('R1', shares) or ('R2', shares)
       up to one of each for each party
 
     Reconstruction takes places in chunks of t+1 values
     """
+    bench_logger = logging.LoggerAdapter(logging.getLogger("benchmark_logger"),
+                                         {"node_id": myid})
 
-    fp = GF(p)
-    secret_shares = [v.value for v in secret_shares]
-    round1_chunks = to_chunks(secret_shares, t + 1)
-    num_chunks = len(round1_chunks)
+    # (optional) Induce faults
 
+    secret_shares = [v.value for v in secret_shares]
     if config is not None and config.induce_faults:
```

Comment:
Moving this chunk right here to execute before assigning `round1_chunks = to_chunks(secret_shares, t+1)` was really the main bugfix of this PR.